processing/src/main/java/org/apache/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -175,10 +175,13 @@ private Sequence<ResultRow> mergeGroupByResults(
       ResponseContext context
   )
   {
+    Sequence<ResultRow> process;
     if (isNestedQueryPushDown(query)) {
-      return mergeResultsWithNestedQueryPushDown(query, resource, runner, context);
+      process = mergeResultsWithNestedQueryPushDown(query, resource, runner, context);
+    } else {
+      process = mergeGroupByResultsWithoutPushDown(query, resource, runner, context);
     }
-    return mergeGroupByResultsWithoutPushDown(query, resource, runner, context);
+    return GroupByQueryRunnerFactory.wrapSummaryRowIfNeeded(query, process);
```

Comment:
I'm surprised that this was required, which test caused you to need this change?  I say this because the only way you should be able to get a completely empty sequence here is if the "leaf nodes" are producing completely empty sequences.  The change in the other place should ensure that no leaf node ever produces a completely empty sequence, meaning that this change shouldn't be necessary...

Reply:
Thank you for taking a look!
unfortunately its needed - I've linked the test(s) checking this.

The leaf nodes are not necessarily aggregating (in case of `distinct`) so an empty sequence may be produced - the merger supposed to aggregate them - that's why this is needed.

For nested query stuff the merge runner becomes this [lambda](https://github.com/apache/druid/blob/d201ea0ece00cf4408b0f358d93071472c606c4e/processing/src/main/java/org/apache/druid/query/groupby/GroupByQueryQueryToolChest.java#L125-L132) (note: I don't know why I didn't placed this call there - just moved it)

example tests
* [testCountDistinctNonApproximateEmptySet](https://github.com/apache/druid/pull/14748/files#diff-fe60470a1d97fa758f70099a5ed49168cfe57f726dc6f08578df9b575b9a21a1R1953) is a sql level one 
* [testSummaryrowForEmptySubqueryInput](https://github.com/apache/druid/pull/14748/files#diff-df810dbf0150a412cc7b71f379267ecd7f1f1f4da906e961d7cbce718147d7b7R13004) as a runnertest